ui/raidboss/data/06-ew/trial/rubicante.ts

```diff
@@ -25,37 +25,55 @@ Locations/Slots:
 00200010 = Arrows rotating CCW
 00400004 = Clear CCW arrows
 
-04 flags:
-00100010 = starts N/S, cleaves NW/SE
-01000100 = starts N/S, cleaves NW/SE (same as above? needs investigation?)
-00020002 = starts N/S, cleaves E/W
+04 flags:  **WIP**
+00100010 = cleaves NW/SE (vfx starts from NW)
+01000100 = cleaves NW/SE (vfx starts from SE - otherwise same as above)
+00020002 = cleaves E/W (vfx starts from W)
+00800080 = cleaves E/W (vfx starts from E - otherwise same as above)
 00080004 = clear indicator
 
 05-0C flags:
 00010004 = initial telegraph - straight line in given direction
 00020004 = initial telegraph - CW jagged line (turns 1 dir CW from starting direction) (only used for Purg 2)
+00080004 = initial telegraph - CCW jagged line (turns 1 dir CCW from starting direction) (only used for Purg 2)
 00800004 = clear telegraph of fire line
 00100004 = fire pulse - straight line in given direction
 00200004 = fire pulse - CW jagged line (turns 1 dir CW from starting direction) (paired with CCW arrows)
 00400004 = fire pulse - CCW jagged line (turns 1 dir CCW from starting direction) (paired with CW arrows)
-
  */
 
 // 7CD4 Ghastly Torch during add phase *is* an aoe but is constant and small, so skipped.
 
 // TO-DO:
-//  - Rewrite Ordeal of Purgation to use Map Effects and identify directional source of cone/safe area(s)
-//  - Flamerake - identify initial direction of line cleave?
+//  - Flamerake - identify initial orientation of line cleave?
 
 export interface Data extends RaidbossData {
+  purgation: number;
+  purgationLoc?: string;
+  purgationLine?: string;
+  purgationMidRotate?: string;
   dualfireTarget: boolean;
 }
 
+const dirs = ['north', 'northeast', 'east', 'southeast', 'south', 'southwest', 'west', 'northwest'];
```

Comment:
I think this code is ok as written, but there's a little bit more testing safety if you write it like https://github.com/quisquous/cactbot/blob/main/ui/raidboss/data/06-ew/raid/p8s.ts#L757-L769 (calling output with literal strings rather than a `output[someVariable]` indirection.  If you use a literal string, tests will verify that there's an output string, but there's no such protection for variables and you have to hope that they line up.